edocumentation-service/src/main/java/org/dcsa/edocumentation/service/unofficial/BookingValidationService.java

```diff
@@ -71,11 +72,90 @@ private String validateBooking(Booking booking) {
     if (isInThePast(today, booking.getExpectedArrivalAtPlaceOfDeliveryEndDate())) {
       return "expectedArrivalAtPlaceOfDeliveryEndDate is in the past";
     }
-
-    return null;
+    return validateShipmentLocations(booking.getShipmentLocations());
   }
 
   private boolean isInThePast(LocalDate today, LocalDate time) {
     return time != null && today.isAfter(time);
   }
+
+  private String validateShipmentLocations(Set<ShipmentLocation> shipmentLocations) {
+    if (shipmentLocations.isEmpty()) {
+      return "Invalid booking: Shipment locations should not be empty";
+    }
+
+    Predicate<LocationType> predicateHasPREorPOL =
+        code -> code == LocationType.PRE || code == LocationType.POL;
+
+    Predicate<LocationType> predicateHasPODorPDE =
+        code -> code == LocationType.POD || code == LocationType.PDE;
```

Comment:
Nit: These could be `private static final Predicate<ShipmentLocation> HAS_SOURCE_LOCATION` and `private static final Predicate<ShipmentLocation> HAS_DESTINATION_LOCATION`

The use of `predicate` in the name is Hungarian notation and we generally try to avoid doing that (the `date time` suffix being a notable exception).

Reply:
Makes sense, thanks for the explainer. 
I used the suggested notation.